Libraries/Components/ScrollView/ScrollView.js

```diff
@@ -549,24 +552,31 @@ export type Props = $ReadOnly<{|
  * multiple columns, infinite scroll loading, or any number of other features it
  * supports out of the box.
  */
-const ScrollView = createReactClass({
-  displayName: 'ScrollView',
-  mixins: [ScrollResponder.Mixin],
-
-  _scrollAnimatedValue: (new AnimatedImplementation.Value(
+class ScrollView extends React.Component<Props, State> {
+  _createScrollResponder = () => {
+    const scrollResponder = {...ScrollResponder.Mixin};
+    for (const key in ScrollResponder.Mixin) {
+      if (typeof scrollResponder[key] === 'function') {
+        scrollResponder[key] = scrollResponder[key].bind(this);
+        (this: any)[key] = scrollResponder[key].bind(this);
```

Comment:
Turning this to class component turned out pretty straightforward. However, problem with binding `scrollResponder` to `ScrollView` is that methods from mixin will refer to `ScrollView` context, while they were not assigned to the ScrollView instance, but to `this._scrollResponder`. To mitigate that, I added a hack that copies methods from mixin back to the `ScrollView`. Wonder if it makes sense to actually do only that and only put lifecycle methods (componentWillMount/componentWillUnmount) inside `this._scrollResponder`

Reply:
Yes, I wanted to address that, but I left it in a, let's say, "request for comments" state and lost track on this issue – this was the reason I left the `this._scrollResponder` so we can differentiate on state and lifecycle of mixin vs component. This is what I meant while writing:

> Wonder if it makes sense to actually do only that and only put lifecycle methods (componentWillMount/componentWillUnmount) inside this._scrollResponder

So I think it's gonna be safer to revert this change you landed locally and get this sorted here (unless you'd like to to fix it on your own while landing it, then be my guest :D)